activiti-services/activiti-services-events/src/test/java/org/activiti/services/events/MessageProducerListenerTest.java

```diff
@@ -0,0 +1,88 @@
+package org.activiti.services.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.activiti.bpmn.model.BpmnModel;
+import org.activiti.bpmn.model.EndEvent;
+import org.activiti.bpmn.model.Process;
+import org.activiti.bpmn.model.SequenceFlow;
+import org.activiti.bpmn.model.StartEvent;
+import org.activiti.engine.ProcessEngine;
+import org.activiti.engine.ProcessEngineConfiguration;
+import org.activiti.services.api.events.ProcessEngineEvent;
+import org.activiti.services.events.tests.util.MockMessageChannel;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = MessageProducerListenerTest.ContextConfig.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE,
+        classes = {MessageProducerListenerTest.class, MessageProducerActivitiEventListener.class, MessageProducerCommandContextCloseListener.class})
+public class MessageProducerListenerTest {
+
+    @Autowired
+    private MessageProducerActivitiEventListener eventListener;
+
+    @Configuration
+    @ComponentScan({"org.activiti.services.events.tests.util", "org.activiti.services.events.converter", "org.activiti.services.core.model.converter"})
+    public class ContextConfig {
+    }
+
+    @Test
+    public void executeListener() {
```

Comment:
Do you think that it would be easy to add another test to illustrate the scenario where the transaction has been rolled back and then no events are produced? If it's easy, please do that in this pull request. If it's complex let's do it later on.